widgetssdk/src/main/java/com/glia/widgets/engagement/end/domain/LocallyEndEngagementUseCase.kt

```diff
@@ -0,0 +1,16 @@
+package com.glia.widgets.engagement.end.domain
+
+import com.glia.widgets.engagement.end.EndEngagement
+import com.glia.widgets.engagement.end.EngagementEndReasonDataSource
+
+internal class LocallyEndEngagementUseCase(
```

Comment:
These two usecases are identical all except reason. Maybe join them and call the usecase with an argument?